serverlessworkflow/modules/ROOT/pages/cloud/enhanced-service-discovery.adoc

```diff
@@ -0,0 +1,129 @@
+to perform http requests, the following sketch demonstrate how the SWUrl is composed:
+
+[source,shell]
+----
+kubernetes:<group>/<version>/<kind>/<namespace>/<name>?<attributeName>=<attributeValue>
+\________/ \_____/ \_______/ \____/ \_________/ \____/ \______________________________/
+  scheme    group   version   kind   namespace   name   additional resource attributes
+            \____________________/                       \__________________________/
+                Kubernetes GVK                                  Supported value: port_name
+----
+
+The following scheme values are supported:
+
+* **kubernetes**
+* **openshift**
+* **knative**
+
+For the GVK, the following Kubernetes Resources are currently supported:
+
+* **v1/service**
+* **serving.knative.dev/v1/service**
+* **v1/pod**
+* **apps/v1/deployment**
+* **apps.openshift.io/v1/deploymentconfig**
+* **apps/v1/statefulset**
+* **route.openshift.io/v1/route**
+* **networking.k8s.io/v1/ingress**
+
+[IMPORTANT]
+====
+The **GVK** (Group, Version and Kind) and the **Name** values are **mandatory** and can't be empty, otherwise, the engine will throw a validation issue and the discovery will not be performed.  +
+The **namespace** is **optional**. In case of an empty value, the current namespace or context will be used.
+====
+
+== Configuration
+
+You can configure the Service Discovery with two **build time** properties, which are:
+
+* `quarkus.kogito.sw.discovery.enabled`: Disable the service discovery. Defaults to _true_.
+* `quarkus.kogito.kubernetes-client.okhttp.static.logger`: Weather or not to print the `http`communication between the client and the Kubernetes API. Defaults to _false_.
+**  Note that, this system property can change in the future since this feature will be ported to the _quarkus-kubernetes-client_ extension. For more info please refer to this link:{quarkus_issue_url}[link].
+
+
+== Startup time: discovery enabled x disabled
+
+[tabs]
+====
+Service Discovery Enabled::
++
+[source,shell]
+----
+(main) sw-service-discovery 1.0.0-SNAPSHOT on JVM (powered by Quarkus 999-SNAPSHOT) started in 2.360s. Listening on: http://0.0.0.0:8080
+----
+Service Discovery Disabled::
++
+[source,shell]
+----
+(main) sw-service-discovery 1.0.0-SNAPSHOT on JVM (powered by Quarkus 999-SNAPSHOT) started in 1.507s. Listening on: http://0.0.0.0:8080
+----
+====
+
+[IMPORTANT]
+====
+When using this feature, you need to balance if the delayed startup time is something
+that your application can afford.
+====
+
+== How does it work?
+
+The service discovery is performed at the Quarkus _STATIC_INIT_ time, which means, before the application gets started. It scans the system properties and searches for the _SWUrl_ pattern, if at least one is found, the engine will parse the SWUrl and then query the Kubernetes API for the given resource and override the given system property. Let's see a small example of it:
```

Comment:
```suggestion
During application startup, the service discovery is performed at the Quarkus _STATIC_INIT_ time. First, it scans the Quarkus configuration and searches for the Kubernetes URI pattern. Then if at least one is found, the engine will parse the URI, query the Kubernetes API for the given resource, and override the given system property. Let's see a small example of it:
```
Please link the `Kubernetes URI` to the section you're explaining about the URI structure.